### PR TITLE
Migration to populate the read perms in base entities.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/701/
Test PASSed.

File: internal/charmstore/migrations.go [hunk 1]
```diff
@@ -154,3 +158,32 @@ func createBaseEntities(db StoreDatabase) error {
...
+		}
+		if err := baseEntities.UpdateId(entity.URL, bson.D{{
+			"$set", bson.D{{"acls.read", readPerm}},
+		}}); err != nil {
+			return errgo.Notef(err, "cannot populate ACL for base entity %s", entity.URL)
+		}
+		counter++
+	}
+	if err := iter.Close(); err != nil {
+		return errgo.Notef(err, "cannot populate ACLs for base entities")
```
Comment (rogpeppe) [inline internal/charmstore/migrations.go]: cannot close entity iterator ?

Comment (frankban) [inline internal/charmstore/migrations.go]: Done.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/702/
Test PASSed.

**Comment by urosj** [comment]
:+1: QA

File: internal/charmstore/migrations_test.go [hunk 2]
```diff
@@ -453,6 +473,178 @@ func (s *migrationsSuite) TestCreateBaseEntitiesSomeUpdates(c *gc.C) {
...
+		Name:   "rails",
+		Public: true,
+		ACLs: mongodoc.ACL{
+			Read: []string{params.Everyone, "who"},
+		},
+	})
+}
+
+func (s *migrationsSuite) TestCreateBaseEntitiesAndPopulateReadACL(c *gc.C) {
+	s.patchMigrations(c, getMigrations("base entities creation", "read acl creation"))
```
Comment (jrwren) [inline internal/charmstore/migrations_test.go:529]: NICE!

**Comment by jrwren** [comment]
:+1:

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**